lucet-wasi/src/hostcalls/fs.rs

```diff
@@ -601,7 +576,26 @@ pub fn wasi_path_unlink_file(
     // nix doesn't expose unlinkat() yet
     match unsafe { unlinkat(dir, path_cstr.as_ptr(), 0) } {
         0 => wasm32::__WASI_ESUCCESS,
-        _ => wasm32::errno_from_nix(errno::Errno::last()),
+        _ => {
+            let mut e = errno::Errno::last();
+            // Non-Linux implementations may return EPERM when attempting to remove a
```

Comment:
I realized right after clicking approve: do we know *which* non-Linux `unlinkat` may return EPERM? I would imagine this is a macos-ism, but would like to be clear if we can

Reply:
In fact, this is the expected POSIX behavior. Only Linux returns `EISDIR` for this case, and this is even documented in the Linux man page.

Reply:
Ah, I see that now. Thanks!